CMakeLists.txt

```diff
@@ -118,6 +118,15 @@ set(LIBDDWAF_SOURCE
     ${libddwaf_SOURCE_DIR}/src/transformer/js_decode.cpp
     ${libddwaf_SOURCE_DIR}/src/transformer/common/utf8.cpp
     ${libddwaf_SOURCE_DIR}/src/libcxx-compat/monotonic_buffer_resource.cpp
+    ${libddwaf_SOURCE_DIR}/src/vendor/fmt/format.cc
+    ${libddwaf_SOURCE_DIR}/src/vendor/radixlib/radixlib.c
+    ${libddwaf_SOURCE_DIR}/src/vendor/lua-aho-corasick/ac_fast.cxx
+    ${libddwaf_SOURCE_DIR}/src/vendor/lua-aho-corasick/ac_slow.cxx
+    ${libddwaf_SOURCE_DIR}/src/vendor/lua-aho-corasick/ac.cxx
+    ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/xss.c
+    ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_html5.c
+    ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_xss.c
+    ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_sqli.c
```

Comment:
I'm not quite sold on this. What if you need to compile the third-party projects with different flags in the future? This may also increase build times a bit because tests will not compile these files again.

Reply:
> What if you need to compile the third-party projects with different flags in the future?
We've had these for ages and haven't needed to use different flags, we can always move the specific library back to `third_party` as a separate library, if that is the case.

>This may also increase build times a bit because tests will not compile these files again.
For a while now the tests use `libddwaf_objects` as a dependency so no need to recompile anything.